### PR TITLE
e2e: Refactored Receive Builder to be consistent with other helpers.

### PR DESCRIPTION
**Pull request by bwplotka**
Signed-off-by: Bartlomiej Plotka <bwplotka@gmail.com>

### CODE REVIEW

File: cmd/thanos/receive.go [hunk 1]
```diff
@@ -769,7 +769,7 @@ func (rc *receiveConfig) registerFlag(cmd extkingpin.FlagClause) {
 	rc.refreshInterval = extkingpin.ModelDuration(cmd.Flag("receive.hashrings-file-refresh-interval", "Refresh interval to re-read the hashring configuration file. (used as a fallback)").
 		Default("5m"))
 
-	cmd.Flag("receive.local-endpoint", "Endpoint of local receive node. Used to identify the local node in the hashring configuration.").StringVar(&rc.endpoint)
+	cmd.Flag("receive.local-endpoint", "Endpoint of local receive node. Used to identify the local node in the hashring configuration. If it's empty AND hashring configuration was provided, it means thie receive will run in RoutingOnly mode.").StringVar(&rc.endpoint)
```
Comment (esnible) [inline cmd/thanos/receive.go:772]: Typo: "thie"

File: test/e2e/e2ethanos/services.go [hunk 2]
```diff
@@ -357,112 +357,92 @@ func (q *QuerierBuilder) collectArgs() ([]string, error) {
...
-// It is configured both for ingesting samples and routing samples to other receivers.
-func NewRoutingAndIngestingReceiverFromFuture(f e2e.FutureInstrumentedRunnable, replicationFactor int, hashring ...receive.HashringConfig) e2e.InstrumentedRunnable {
-	var localEndpoint string
-	if len(hashring) == 0 {
-		localEndpoint = "0.0.0.0:9091"
-		hashring = []receive.HashringConfig{{Endpoints: []string{localEndpoint}}}
-	} else {
-		localEndpoint = f.InternalEndpoint("grpc")
+func NewReceiveBuilder(e e2e.Environment, name string) *ReceiveBuilder {
+	f := e2e.NewInstrumentedRunnable(e, fmt.Sprintf("receive-%v", name)).
```
Comment (esnible) [inline test/e2e/e2ethanos/services.go:372]: Not `%s`?

**Review by bill3tt — DISMISSED** [review]

File: test/e2e/e2ethanos/services.go [hunk 3]
```diff
@@ -357,112 +357,92 @@ func (q *QuerierBuilder) collectArgs() ([]string, error) {
...
-// NewRoutingAndIngestingReceiverFromFuture creates a Thanos Receive instances from a future service.
-// It is configured both for ingesting samples and routing samples to other receivers.
-func NewRoutingAndIngestingReceiverFromFuture(f e2e.FutureInstrumentedRunnable, replicationFactor int, hashring ...receive.HashringConfig) e2e.InstrumentedRunnable {
-	var localEndpoint string
-	if len(hashring) == 0 {
-		localEndpoint = "0.0.0.0:9091"
-		hashring = []receive.HashringConfig{{Endpoints: []string{localEndpoint}}}
-	} else {
-		localEndpoint = f.InternalEndpoint("grpc")
+func NewReceiveBuilder(e e2e.Environment, name string) *ReceiveBuilder {
```
Comment (douglascamata) [inline test/e2e/e2ethanos/services.go:371]: Why no comment for this and all the other new exported functions? 🤔

**Review by matej-g — DISMISSED** [review]
Looking good, minus the typos and doc comments from others :+1:

File: test/e2e/e2ethanos/services.go (hunk 3 above)
Comment (bwplotka) [reply test/e2e/e2ethanos/services.go:371]: Because I was too lazy - will do (:

Comment (bwplotka) [reply test/e2e/e2ethanos/services.go:371]: No methods here have comments actually, so let's do this in another PR.

**Review by matej-g — APPROVED** [review]

**Pull request merged**